### ISSUE TITLE
Non Resource-Pool Cluster problem

### ISSUE THREAD

**Issue by mcasin** [snobear/ezmomi#51]
Hi, 

First of all thanks for the nice tool, it's amazing!

I have a clustered environment and there are no ResourcePools in that cluster, in fact I cannot create a ResourcePool there. (It's supposed to be a hidden root ResourcePool but it doesn't work)

When I try to clone a template I get the erro:

Error: Unable to find Resource Pool 'Resources'

If I try to list all the ResourcePools I get:

MOID               Name
resgroup-411009    Resources
resgroup-409359    Resources
resgroup-162621    Resources
resgroup-162616    Resources
resgroup-306271    Resources
resgroup-433552    Resources
resgroup-64325     Resources
resgroup-425184    Resources

But Resources doesn't work as ResourcePool name.

Anybody can help?

Thanks in advance

**Comment by snobear** [comment]
Hi @mcasin, looks like a bug with the default Resources pool. I'm able to reproduce it.  Thanks for reporting! - will fix asap.

You can downgrade to 0.3.1 as a workaround for now:

```
pip uninstall ezmomi
pip install ezmomi==0.3.1
```

**Comment by plat00n** [comment]
Hello, I have the same issue.
For information we don't have Custom Ressource Pools, just the default one named "Ressources"

$ ezmomi list --type ResourcePool
ResourcePool list
MOID               Name
resgroup-22    Resources

I tried to mannualy create another ressource pool named " Resources " (to find a workaround)
Bug we got one other error message : 

$ ezmomi list --type ResourcePool
ResourcePool list
MOID            Name
resgroup-22     Resources
resgroup-122    Resources

$ ezmomi clone --hostname mycloneserv --cpus 2 --mem 4 --ips 192.168.1.2
Cloning MyOStemplate to new host mycloneserv with 4096MB RAM...
Traceback (most recent call last):
  File "/usr/local/bin/ezmomi", line 4, in <module>
    cli.cli()
  File "/usr/local/lib/python2.7/site-packages/ezmomi/cli.py", line 30, in cli
    ez.clone()
  File "/usr/local/lib/python2.7/site-packages/ezmomi/ezmomi.py", line 220, in clone
    for device in template_vm.config.hardware.device:
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 547, in __call__
    return self.f(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 370, in _InvokeAccessor
    return self._stub.InvokeAccessor(self, info)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/StubAdapterAccessorImpl.py", line 41, in InvokeAccessor
    return self._pc.RetrieveContents([filterSpec])[0].propSet[0].val
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 553, in <lambda>
    self.f(*(self.args + (obj,) + args), **kwargs)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 362, in _InvokeMethod
    return self._stub.InvokeMethod(self, info, args)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/SoapAdapter.py", line 1226, in InvokeMethod
    obj = SoapResponseDeserializer(outerStub).Deserialize(fd, info.result)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/SoapAdapter.py", line 747, in Deserialize
    self.parser.ParseFile(response)
xml.parsers.expat.ExpatError: not well-formed (invalid token): line 1, column 0

Hoping it will help you to find a solution :)

NB:
ezmomi (0.4.1)
pyvmomi (5.5.0)

**Comment by snobear** [comment]
This has been fixed in ezmomi 0.4.2 which has just been published to pypi.

```
pip install -U ezmomi
````

**Comment by plat00n** [comment]
Hello,

Issue seems still to be here. It don't tell us that the ressource pool is not present but we still have : 

[myuser@myserv ezmomi]$ ezmomi clone --hostname myclonedserv --cpus 2 --mem 4 --ips 192.168.1.2
Cloning T-centos6-OVH to new host myclonedserv with 4096MB RAM...
Traceback (most recent call last):
  File "/usr/local/bin/ezmomi", line 4, in <module>
    cli.cli()
  File "/usr/local/lib/python2.7/site-packages/ezmomi/cli.py", line 30, in cli
    ez.clone()
  File "/usr/local/lib/python2.7/site-packages/ezmomi/ezmomi.py", line 220, in clone
    for device in template_vm.config.hardware.device:
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 547, in __call__
    return self.f(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 370, in _InvokeAccessor
    return self._stub.InvokeAccessor(self, info)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/StubAdapterAccessorImpl.py", line 41, in InvokeAccessor
    return self._pc.RetrieveContents([filterSpec])[0].propSet[0].val
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 553, in <lambda>
    self.f(*(self.args + (obj,) + args), **kwargs)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/VmomiSupport.py", line 362, in _InvokeMethod
    return self._stub.InvokeMethod(self, info, args)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/SoapAdapter.py", line 1226, in InvokeMethod
    obj = SoapResponseDeserializer(outerStub).Deserialize(fd, info.result)
  File "/usr/local/lib/python2.7/site-packages/pyVmomi/SoapAdapter.py", line 747, in Deserialize
    self.parser.ParseFile(response)
xml.parsers.expat.ExpatError: not well-formed (invalid token): line 1, column 0

NB:
$ ezmomi list --type ResourcePool
ResourcePool list
MOID           Name
resgroup-67    Resources

and 

ezmomi (0.4.2)
pyvmomi (5.5.0)

Thanks for the help :)

ps : is ezmomi is working with pyvmomi 5.5.0.2014.1.1 (which is the latest one available in pypi)
